SpatialGDK/Source/SpatialGDK/Public/Utils/EntityFactory.h

```diff
@@ -25,12 +25,39 @@ struct RPCsOnEntityCreation;
 using FRPCsOnEntityCreationMap = TMap<TWeakObjectPtr<const UObject>, RPCsOnEntityCreation, FDefaultSetAllocator,
 									  TWeakObjectPtrMapKeyFuncs<TWeakObjectPtr<const UObject>, SpatialGDK::RPCsOnEntityCreation, false>>;
 
+struct EntityComponents
```

Comment:
This looks like a builder pattern but without the building. Could we lean into the builder idea for entity construction? Call this entity builder, with all your mutable components while building, then the build() call returns the Entity representation (list of components, whatever it is) rather than the same object so it's obvious one has mutable components and the other doesn't

Reply:
Hm, yeah, I was considering a builder the entire time, so maybe I should just give in to the dark side.